console/frontend/src/main/frontend/src/app/views/monitors/monitors.component.html

```diff
@@ -46,216 +47,270 @@ <h4>
-            </button>
-            <button
-              appHasAccessToLink="deleteMonitor"
-              *ngIf="!monitor.raised && monitor.edit"
-              title="Delete Monitor {{ monitor.name }}"
-              class="btn btn-danger btn-xs space-it-out"
-              type="button"
-              (click)="deleteMonitor(monitor)"
-            >
-              <i class="fa fa-close"></i> Delete
-            </button>
-            <button
-              appHasAccessToLink="updateMonitor"
-              *ngIf="!monitor.raised && !monitor.edit"
-              title="Edit Monitor {{ monitor.name }}"
-              class="btn btn-info btn-xs space-it-out"
-              type="button"
-              (click)="monitor.edit = true"
-            >
-              <i class="fa fa-pencil"></i> Edit
-            </button>
-            <button
-              appHasAccessToLink="updateMonitor"
-              *ngIf="!monitor.raised && monitor.edit"
-              title="Save Monitor {{ monitor.name }}"
-              class="btn btn-primary btn-xs space-it-out"
-              type="button"
-              (click)="edit(monitor)"
-            >
-              <i class="fa fa-save"></i> Save
-            </button>
-            <button
-              title="Download Monitor Configuration"
-              class="btn btn-secondary btn-xs"
-              type="button"
-              (click)="downloadXML(monitor.name)"
-            >
-              <i class="fa fa-file-code-o"></i> XML
-            </button>
-          </div>
-          <div class="m-l-n m-r-n almost-full-width">
-            <div class="col-lg-5 col-md-8 m-r m-t-n-xs">
-              <h4>
-                {{ monitor.displayName }}
-                <span class="m-l-sm badge badge-info">{{ monitor.type }}</span>
-              </h4>
-            </div>
-            <div class="col-lg-3 hidden-md" *ngIf="monitor.lastHit">
-              Last Hit: <span appToDate [time]="monitor.lastHit"></span>
-            </div>
-            <div class="col-lg-2 hidden-md" *ngIf="monitor.hits">Hits: {{ monitor.hits }} since last raise</div>
-          </div>
-        </div>
-        <div class="ibox-content">
-          <div *ngIf="monitor.raised" role="alert">
-            <div class="col-md-6">
-              Source:
-              {{ monitor.alarm.source ? monitor.alarm.source : 'Manually Triggered' }}<br />
-              Severity: {{ monitor.alarm.severity }}<br />
-            </div>
-          </div>
-          <div *ngIf="monitor.edit">
-            <form role="form" (submit)="edit(monitor)" appQuickSubmitForm>
-              <div class="form-group">
-                <label class="col-sm-3 control-label">Name</label>
-                <div class="col-sm-9">
-                  <input class="form-control m-b" name="name" type="text" [(ngModel)]="monitor.displayName" />
-                </div>
+  @if (monitors.length > 0) {
+    <div class="row">
+      @for (monitor of monitors; track monitor.name) {
```

Comment:
Couldn't two monitors have the same name in the 'All' tab?

Reply:
I'm not sure if you can, but you shouldn't right? :)

Reply:
Ofc you shouldn't 👀 

You never know, I've seen an integration with multiple configurations that all have the same boilerplate. I wouldn't know if it also included monitors tho.